ori/volume/server/volume_create.go

```diff
@@ -17,260 +17,92 @@ package server
 import (
 	"context"
 	"fmt"
-	"strings"
 
 	"github.com/go-logr/logr"
-	snapshotv1 "github.com/kubernetes-csi/external-snapshotter/client/v6/apis/volumesnapshot/v1"
-	volumev1alpha1 "github.com/onmetal/cephlet/ori/volume/api/v1alpha1"
-	"github.com/onmetal/cephlet/ori/volume/apiutils"
-	"github.com/onmetal/controller-utils/metautils"
-	storagev1alpha1 "github.com/onmetal/onmetal-api/api/storage/v1alpha1"
+	"github.com/onmetal/cephlet/pkg/api"
+	"github.com/onmetal/cephlet/pkg/limits"
+	"github.com/onmetal/cephlet/pkg/round"
 	ori "github.com/onmetal/onmetal-api/ori/apis/volume/v1alpha1"
-	onmetalimage "github.com/onmetal/onmetal-image"
-	"github.com/onmetal/onmetal-image/oci/image"
-	"github.com/onmetal/onmetal-image/oci/remote"
-	corev1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/api/errors"
-	"k8s.io/apimachinery/pkg/api/resource"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/types"
-	"k8s.io/utils/pointer"
-	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-func GetSanitizedImageName(image string) string {
-	image = strings.ReplaceAll(image, "/", "-")
-	image = strings.ReplaceAll(image, ":", "-")
-	return strings.ReplaceAll(image, "@", "-")
+type CephVolumeConfig struct {
+	image *api.Image
 }
 
-func getImageSize(ctx context.Context, imageName string) (*resource.Quantity, error) {
-	reg, err := remote.DockerRegistry(nil)
-	if err != nil {
-		return nil, fmt.Errorf("failed to initialize registry: %w", err)
-	}
-
-	img, err := reg.Resolve(ctx, imageName)
-	if err != nil {
-		return nil, fmt.Errorf("failed to resolve image ref in registry: %w", err)
-	}
-
-	layers, err := img.Layers(ctx)
-	if err != nil {
-		return nil, fmt.Errorf("failed to get layers for image: %w", err)
-	}
+func (s *Server) getCephVolumeConfig(ctx context.Context, log logr.Logger, volume *ori.Volume) (*CephVolumeConfig, error) {
+	log.V(2).Info("Getting ceph volume config")
+	defer log.V(2).Info("Successfully got ceph volume config")
```

Comment:
Remove the `defer` - if you're returning an `error`, the log would still be printed which doesn't make much sense.